pkg/provider/bitbucketdatacenter/parse_payload.go

```diff
@@ -155,6 +155,11 @@ func (v *Provider) ParsePayload(_ context.Context, _ *params.Run, request *http.
 		}
 		v.pullRequestNumber = e.PullRequest.ID
 	case *types.PushRequestEvent:
+		for _, change := range e.Changes {
+			if change.ToHash == "0000000000000000000000000000000000000000" {
+				return nil, nil
+			}
+		}
```

Comment:
```suggestion
		for _, change := range e.Changes {
			if change.ToHash == "0000000000000000000000000000000000000000" && change.Type == "DELETE {
				return nil, nil
			}
		}
```


@infernus01 if you see in payload changes have `type: DELETE` so it would be good if you add this condition with `&&`
```
"changes": [{
        "ref": {
            "id": "refs/heads/test-26196",
            "displayId": "test-26196",
            "type": "BRANCH"
        },
        "fromHash": "abda6907a37c451017acea247ee22d2dc457e1aa",
        "toHash": "0000000000000000000000000000000000000000",
        "refId": "refs/heads/test-26196",
        "type": "DELETE"
    }]
```

Reply:
I think that change to make is for you @infernus01 :)

Reply:
PTAL